### PR TITLE
Add baseball model from Doing Bayesian Data Analysis

### PR DESCRIPTION
**Pull request by IamGianluca**
This Pull Request introduced the `baseball.stan` model from chapter 9 of Doing Bayesian Data Analysis. I received the approval from John Kruschke for publishing the Stan translated models into this repository. 

As part of this Pull Request I've also attached two other models, named `therapeutic_touch_bernoulli.stan` and `therapeutic_touch_binomial.stan`. I haven't pushed the final version of the Jupyter notebook for this model yet, as I want to gather some feedback from the baseball one first.

I think a notebook like this could be greatly improved the readability of the model, especially for user with not much expertise in Bayesian Inference.

As I am myself learning Bayesian Inference while reading DBDA, I'm planning to make some further adjustments to the notebook in the future weeks.

### CODE REVIEW

**Comment by bob-carpenter** [comment]
Can you get that in writing or ask Krushcke to put some license on them, like BSD, that lets anyone use them for anything with attribution?

**Comment by IamGianluca** [comment]
Hi Bob, I've just asked to John to comment on this Pull Request.

**Comment by bob-carpenter** [comment]
I just followed up, but I didn't have your email to cc you on the mail to John Kruschke.  Hopefully he can just post licensing terms of the code he is distributing.

File: DBDA/ch9/baseball.stan [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    kappa0 = kappaMinusTwo0 + 2;
+    alpha0 = omega0 * (kappa0 - 2) + 1;
+    beta0 = (1 - omega0) * (kappa0 - 2) + 1;
+
+    for (c in 1:N_positions) {
+        kappa[c] = kappaMinusTwo[c] + 2;
+        alpha[c] = omega[c] * (kappa[c] - 2) + 1;
+        beta[c] = (1 - omega[c]) * (kappa[c] - 2) + 1;
+    }
```
Comment (bob-carpenter) [inline DBDA/ch9/baseball.stan]: If you make these vectors you can vectorize this oop, so it looks like:

```
kappa = kappaMinusTwo + 2;
alpha = omega .* (kappa - 2) + 1;
beta = (1 - omega) .* (kappa - 2) + 1.
```

But usually you cold just do `beta = kappa - alpha`, but I can't follow all this plus and minus business.  Presumably this is all done to match something in Kruschke's book.  You could do this all more directly in Stan.

File: DBDA/ch9/baseball.stan [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    real<lower=0> kappa0;
+    real<lower=0> alpha0;
+    real<lower=0> beta0;
+    real<lower=0> kappa[N_positions];
+    real<lower=0> alpha[N_positions];
+    real<lower=0> beta[N_positions];
+
+    kappa0 = kappaMinusTwo0 + 2;
+    alpha0 = omega0 * (kappa0 - 2) + 1;
+    beta0 = (1 - omega0) * (kappa0 - 2) + 1;
```
Comment (bob-carpenter) [inline DBDA/ch9/baseball.stan]: Use `kappaMinusTwo0` instead of `kappa0 - 2`

File: DBDA/ch9/baseball.stan [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    }
+}
+model {
+    omega0 ~ beta(A, B);
+    kappaMinusTwo0 ~ gamma(S, R);
+    
+    for (c in 1:N_positions) {
+        omega[c] ~ beta(alpha0, beta0);
+        kappaMinusTwo[c] ~ gamma(S, R);
+    }
```
Comment (bob-carpenter) [inline DBDA/ch9/baseball.stan]: vectorize all this to:

```
omega ~ beta(alpha0, beta0);
kappaMinusTwo ~ gamma(S, R);
```

File: DBDA/ch9/baseball.stan [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    
+    for (i in 1:P) {
+        theta[i] ~ beta(alpha[positions[i]], beta[positions[i]]);
+        y[i] ~ binomial(N[i], theta[i]);
+    }
```
Comment (bob-carpenter) [inline DBDA/ch9/baseball.stan]: vectorize this to:

```
theta ~ beta(alpha[positions], beta[positions]);
y ~ binomial(n, theta);
```

**Comment by bob-carpenter** [comment]
We can't merge this because we don't have permission to use Kruschke's code, so I'm closing.

**Comment by IamGianluca** [comment]
Thank you so much for the feedback Bob. I really appreciate it! I'm going to make the changes you suggested.

**Comment by bob-carpenter** [comment]
I'm happy to provide feedback for the models you write for
the book even if they're not going on our site.  Just let me
know where they're at to comment.

- Bob

**Pull request closed without merging**